src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -281,8 +285,11 @@ public int order() {
                 return;
             } else {
                 auditLog.logMissingPrivileges(action, request, task);
-                log.debug("no permissions for {}", pres.getMissingPrivileges());
-                listener.onFailure(new ElasticsearchSecurityException("no permissions for " + pres.getMissingPrivileges()+ " and "+user, RestStatus.FORBIDDEN));
+                String err = (injectedRoles == null) ?
+                        String.format("no permissions for %s and %s" , pres.getMissingPrivileges(), user) :
+                        String.format("no permissions for %s and associated roles %s ", pres.getMissingPrivileges(), injectedRoles);
```

Comment:
nit: trim extra space in the error message.

Reply:
Good catch. Thanks